plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts

```diff
@@ -298,29 +301,34 @@ export class AwsLambdaInstrumentation extends InstrumentationBase {
   }
 
   private static _determineParent(
-    httpHeaders: APIGatewayProxyEventHeaders
+    httpHeaders: APIGatewayProxyEventHeaders,
+    disableAwsPropagation: boolean
   ): OtelContext {
     let parent: OtelContext | undefined = undefined;
-    const lambdaTraceHeader = process.env[traceContextEnvironmentKey];
-    if (lambdaTraceHeader) {
-      parent = awsPropagator.extract(
-        otelContext.active(),
-        { [AWSXRAY_TRACE_ID_HEADER]: lambdaTraceHeader },
-        headerGetter
-      );
-    }
-    if (parent) {
-      const spanContext = trace.getSpan(parent)?.spanContext();
-      if (
-        spanContext &&
-        (spanContext.traceFlags & TraceFlags.SAMPLED) === TraceFlags.SAMPLED
-      ) {
-        // Trace header provided by Lambda only sampled if a sampled context was propagated from
-        // an upstream cloud service such as S3, or the user is using X-Ray. In these cases, we
-        // need to use it as the parent.
-        return parent;
+
+    if (!disableAwsPropagation) {
+      const lambdaTraceHeader = process.env[traceContextEnvironmentKey];
+      if (lambdaTraceHeader) {
+        parent = awsPropagator.extract(
+          otelContext.active(),
+          { [AWSXRAY_TRACE_ID_HEADER]: lambdaTraceHeader },
+          headerGetter
+        );
+      }
+      if (parent) {
+        const spanContext = trace.getSpan(parent)?.spanContext();
+        if (
+          spanContext &&
```

Comment:
Sorry to have this idea this late but does it work to add a check that that span context is valid and no need for the option? I just realized that I think in the problematic case the context has sampled = false, but actually doesn't have a valid span ID either and that could be enough of a signal

Reply:
What do you mean by checking that the span context is valid? Doesn't the `if (spanContext ...` check covers it?
Anyway, for our usage, we'd be interested to have this ability even if x-ray is turned on, I'll try to describe the scenario, hope it'll be clear:

Consider the following:

![image](https://user-images.githubusercontent.com/16322616/123540961-55aec180-d74a-11eb-90ec-4c10d5b829cf.png)

* Mircrosevice sends span with id `x` to the collector
* API Gateway sends span with id `y` to x-ray
* Lambda sends span with id `z` and parent span id of `y`.

We end up only having spans `x` and `z` in the collector, resulting in a broken trace.

Reply:
I think if spanContext is only in the case that there is no span in the context, in practice I don't know if it can actually happen with this code.

There is an isSpanContextValid function

https://www.github.com/open-telemetry/opentelemetry-js-api/tree/main/src%2Fapi%2Ftrace.ts


For that use case, is the idea for the customer to have the trace both in x-ray and another backend via the collector and show the same trace in them?

Reply:
In this use case, the customer is interested only in having `x` and `z` where the parentSpanId of z is x.

Reply:
In that case they can disable x-ray right? Is there a reason to enable it and be charged even without needing y?

Reply:
Some of our customers may have an existing x-ray set-up and may want to use our product in addition to other usages, without integrating x-ray traces into our product.  
Our product provides some other value on top of observability.  

Anyway, IMO it's up for the user to decide, each one has different considerations and it's reasonable to let the user decide how to configure this